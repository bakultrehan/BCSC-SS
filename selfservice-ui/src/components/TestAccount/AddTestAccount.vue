/** * Add TestAccount component */

<template>
  <v-card class="mx-auto " flat="">
    <v-item-group mandatory>
      <v-container>
        <v-row class="mx-4">
          <v-col cols="12" flat>
            <v-card flat>
              <div
                class="text-left"
                v-html="$t('AddTestAccount.pageinfo')"
              ></div>
            </v-card>
          </v-col>

          <v-col class="col-12" v-if="errorStatus || successStatus">
            <Alert
              type="error"
              v-if="errorStatus"
              class="alert-top text-left"
              >{{ $t('AddTestAccount.errorMessage') }}</Alert
            >
            <Alert
              type="success"
              class="alert-top text-left"
              v-if="successStatus"
            >
              {{ successCount.created }}
              {{ $t('AddTestAccount.successCreatedMessage') }}
              <br />
              {{ successCount.skipped }}
              {{ $t('AddTestAccount.successSkippedMessage') }}
            </Alert>
          </v-col>
          <v-col cols="12" flat>
            <v-card flat>
              <TextArea
                v-model="testAccounts"
                :label="$t('AddTestAccount.special_notes')"
                type="text"
                outlined
                rows="10"
                name="test-account-text"
                id="test-account-text"
                data-test-id="test-account-text"
              />
            </v-card>
          </v-col>
        </v-row>
      </v-container>
    </v-item-group>
    <v-col cols="12">
      <v-card flat>
        <v-divider></v-divider>
        <v-card-actions class="mx-4">
          <v-spacer></v-spacer>

          <Button
            :disabled="testAccounts === ''"
            :loading="isLoading"
            class="white--text submit-test-account ml-6"
            depressed
            @click="submitTestAccount"
            name="btn-test-account"
            data-test-id="btn-test-account"
            >{{ $t('AddTestAccount.btnSaveChanges') }}</Button
          >
        </v-card-actions>
      </v-card>
    </v-col>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Button from '@/Atomic/Button/Button.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Alert from '@/Atomic/Alert/Alert.vue';

const TestAccountModule = namespace('TestAccountModule');

@Component({
  components: {
    Button,
    TextArea,
    Alert,
  },
})
export default class TestAccount extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @TestAccountModule.Action('addTestAccounts')
  public addTestAccounts!: any;
  @TestAccountModule.Getter('isLoading')
  public isLoading!: boolean;

  @TestAccountModule.Action('clearStatus') public clearStatus!: any;
  @TestAccountModule.Getter('getTestAccountSuccessData')
  public getTestAccountSuccessData!: any;

  private testAccounts: string = '';
  private successCount: any = { created: 0, skipped: 0 };
  private successStatus: boolean = false;
  private errorStatus: boolean = false;

  @Watch('getTestAccountSuccessData')
  private ongetTestAccountSuccessDataChanged(val: any) {
    const { successStatus, errorStatus, testAccountSuccessData } = val;
    this.clearAllStatus();
    if (successStatus) {
      this.successStatus = true;
      this.successCount = testAccountSuccessData;
    } else if (errorStatus) {
      this.errorStatus = true;
    }
    setTimeout(() => {
      this.clearAllStatus();
      this.clearStatus();
    }, 10000);
  }

  private submitTestAccount() {
    this.addTestAccounts({
      testAccounts: this.testAccounts,
    });
  }
  private clearAllStatus() {
    this.errorStatus = false;
    this.successStatus = false;
  }

  private mounted() {
    this.clearAllStatus();
    this.clearStatus();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
</style>
