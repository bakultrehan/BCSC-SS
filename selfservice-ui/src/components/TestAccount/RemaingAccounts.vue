/** * RemaingAccounts component */

<template>
  <div
    class="text-center remaining-acc-txt"
    v-if="getTestAccountCount && getTestAccountCount.total"
  >
    <span class="count" :class="getTestAccountCount.warning && 'not-available'">
      {{ getTestAccountCount.available }}</span
    ><span class="divide">/</span
    ><span class="total">{{ getTestAccountCount.total }}</span>
    {{ $t('dashboard.txtTestAccountRemain') }}
  </div>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';

const TestAccountModule = namespace('TestAccountModule');

@Component
export default class RemaingAccounts extends Vue {
  @TestAccountModule.Getter('getTestAccountCount')
  public getTestAccountCount!: any;
  @TestAccountModule.Action('loadTestAccountCount')
  public loadTestAccountCount!: any;

  private mounted() {
    this.loadTestAccountCount();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.remaining-acc-txt {
  color: $BCgovFontColorInvertedDark;
  & .count {
    font-size: 24px;
    margin: 2px;
    font-weight: bold;
    color: $succssGreen;
    &.not-available {
      color: $BCgovInputError;
    }
  }
  & .total {
    margin: 2px;
    font-size: 20px;
    font-weight: bold;
  }
  & .divide {
    font-size: 20px;
    font-weight: bold;
  }
}
</style>
