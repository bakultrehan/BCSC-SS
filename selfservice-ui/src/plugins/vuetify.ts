import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
