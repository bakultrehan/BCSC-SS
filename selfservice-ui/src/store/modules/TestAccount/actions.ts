import { ActionTree } from 'vuex';
import { TestAccountState } from './types';
import { RootState } from '../../types';
import { TestAccounts } from '@/services/TestAccounts';

/**
 * shared Actions
 *
 */
export const actions: ActionTree<TestAccountState, RootState> = {
  /**
   * @param  {} {commit}
   * @param  {} flag redirectfromsummary page
   */
  async addTestAccounts({ commit, dispatch }, data: any) {
    commit('SET_LOADING', true);
    dispatch('clearStatus');
    try {
      const { testAccounts } = data;
      await TestAccounts.addTestAccounts({ testAccounts });
      commit('SET_SUCESS', true);
      commit('SET_ERROR', false);
      commit('SET_LOADING', false);
    } catch {
      commit('SET_ERROR', true);
      commit('SET_SUCESS', false);
      commit('SET_LOADING', false);
    }
  },
  /**
   * clear message
   * @param {*} { commit }
   */
  async clearStatus({ commit }) {
    commit('SET_SUCESS', false);
    commit('SET_ERROR', false);
    commit('SET_LOADING', false);
  }
};
