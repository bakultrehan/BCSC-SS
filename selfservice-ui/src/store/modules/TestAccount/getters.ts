import { GetterTree } from 'vuex';
import { TestAccountState } from './types';
import { RootState } from '../../types';

/**
 * Getters Shared
 */
export const getters: GetterTree<TestAccountState, RootState> = {
  /**
   * isLoading  status
   * @param {*} state
   * @returns {boolean} - loading status
   */
  isLoading(state) {
    return state.isLoading;
  },
  successStatus(state) {
    return state.successStatus;
  },
  errorStatus(state) {
    return state.errorStatus;
  },
  getTestAccountCount(state) {
    return state.testAccountCount;
  },
  getTestAccountSuccessData(state) {
    return {
      testAccountSuccessData: state.testAccountSuccessData,
      errorStatus: state.errorStatus,
      successStatus: state.successStatus,
    };
  },
};
