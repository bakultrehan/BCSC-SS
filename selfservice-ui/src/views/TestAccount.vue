<template>
  <v-card flat>
    <TestAccount />
  </v-card>
</template>

<script>
// @ is an alias to /src
import { Component, Vue } from 'vue-property-decorator';
import TestAccount from '@/components/TestAccount/TestAccount.vue';

@Component({
  components: {
    TestAccount
  }
})
// Home goes here
export default class TestAccountPage extends Vue {}
</script>
