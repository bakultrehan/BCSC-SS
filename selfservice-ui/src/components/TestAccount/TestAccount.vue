/** * TestAccount component */

<template>
  <v-card class="mx-auto outer-card">
    <v-toolbar flat class="bc-subtitle padding-0" dark>
      <v-toolbar-title>{{ $t('testAccountList.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-container>
      <v-row class="ma-5">
        <v-col cols="12" v-if="isLoading">
          <Loading />
        </v-col>

        <v-col
          cols="4"
          class="d-flex align-end flex-column-reverse mb-2"
          v-show="!isLoading"
        >
          <!-- <ProjectActions :id="id" /> -->
        </v-col>

        <v-col cols="12">
          <v-tabs slider-color="d-none" v-model="selectedTab">
            <v-tab class="font-weight-bold">
              {{ $t('testAccountList.tabAddtestAccount') }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{ $t('testAccountList.tabProjectUsage') }}
            </v-tab>
            <v-tab class="font-weight-bold">
              {{ $t('testAccountList.tabViewCardInfo') }}
            </v-tab>
            <div class="remaining-acc"><RemaingAccounts /></div>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <AddTestAccount />
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  {{ $t('testAccountList.tabProjectUsage') }}
                  <!-- <TeamRoles :id="id" /> -->
                </v-card-text>
              </v-card>
            </v-tab-item>

            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>{{ $t('testAccountList.tabViewCardInfo') }}</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Security</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Communications</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Agreements</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
            <v-tab-item class="custom-tabs-items">
              <v-card flat>
                <v-card-text>
                  <p>Prod access request</p>
                </v-card-text>
              </v-card>
            </v-tab-item>
          </v-tabs>
        </v-col>
      </v-row>
    </v-container>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
// import Button from '@/Atomic/Button/Button.vue';
// import TextArea from '@/Atomic/TextArea/TextArea.vue';
// import Alert from '@/Atomic/Alert/Alert.vue';

import AddTestAccount from '@/components/TestAccount/AddTestAccount.vue';
import RemaingAccounts from '@/components/TestAccount/RemaingAccounts.vue';

const TestAccountModule = namespace('TestAccountModule');

@Component({
  components: {
    AddTestAccount,
    RemaingAccounts,
  },
})
export default class TestAccount extends Vue {
  @TestAccountModule.Getter('isLoading')
  public isLoading!: boolean;
  private selectedTab: number = 0;
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.custom-tabs-items {
  border: 1px solid $gray5;
  min-height: 300px; // temp fix
}
.v-tab {
  color: $BCgovFontColorInvertedDark !important;
}
.v-tab--active {
  border: 1px solid $gray5;
  background-color: $BCgovGold5;
  color: $BCgovWhite !important;
}
.tab-headline {
  color: $BCgovFontColorInvertedDark;
}
.remaining-acc {
  width: 51%;
  text-align: right;
}
</style>
