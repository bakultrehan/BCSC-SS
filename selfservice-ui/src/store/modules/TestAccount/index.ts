import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { TestAccountState } from './types';
import { RootState } from '../../types';

/**
 * Sharedmodule state
 */
export const state: TestAccountState = {
  testAccounts: {},
  isLoading: false,
  successStatus: false,
  errorStatus: false,
  testAccountCount: {},
};

const namespaced: boolean = true;

const SharedModule: Module<TestAccountState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};

export default SharedModule;
