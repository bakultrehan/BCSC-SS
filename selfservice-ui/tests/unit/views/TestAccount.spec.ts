import { shallowMount, createLocalVue } from '@vue/test-utils';
import TestAccount from '@/views/TestAccount.vue';
import vuetify from 'vuetify';

describe('TestAccount.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(TestAccount, {
      localVue
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
