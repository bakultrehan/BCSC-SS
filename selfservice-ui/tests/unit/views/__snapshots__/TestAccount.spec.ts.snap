// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestAccount.vue renders props when passed 1`] = `
<v-card-stub
  flat="true"
  loaderheight="4"
  tag="div"
>
  <testaccount-stub
    id="0"
  />
</v-card-stub>
`;
